src/sql/src/plan/statement/ddl.rs

```diff
@@ -318,13 +319,19 @@ pub fn plan_create_source(
         })
     };
 
+    // Convert from sql-parser Compression to dataflow-types Compression
+    let compression: Compression = compression.clone().into();
     let mut with_options = normalize::options(with_options);
 
     let mut consistency = Consistency::RealTime;
     let mut ts_frequency = Duration::from_secs(1);
 
     let (external_connector, mut encoding) = match connector {
         Connector::Kafka { broker, topic, .. } => {
+            if compression != Compression::None {
+                bail!("only NONE compression supported in KAFKA connectors");
```

Comment:
If we make `compression` a field on individual connectors, we won't need these checks.

Reply:
This check isn't related to the presence of the `compression` field in the individual connectors. In fact, the `compression` field is already on individual connectors (just the file one atm).

If I remove this check then a user specifying `COMPRESSION GZIP` with an incompatible connector will get a successful query, but that option will be ignored which would be surprising. Did I get what you mean correctly?

Reply:
I meant making it part of the connectors in the SQL AST. What I mean is that `compression` is a top-level field in `CreateSourceStatement` -- I am suggesting that we remove that, and have it only be a field on the connector descriptions that are part of that statement (i.e., `sql-parser::ast::defs::ddl::Connector::File`).

As I said above, we should also move compression into the connector-specific part of the syntax (like kafka topic, file path, s3 bucket, etc.) so that it is impossible to specify `COMPRESSION GZIP` for an incompatible connector.